### DISCUSSION TITLE
v1.6-experimental1

### DISCUSSION THREAD

**Discussion by duncanmacmichael** [microsoft/WindowsAppSDK#4470] [Announcements]
## 1.6 Experimental 1 (1.6.240531000-experimental1) 🧪

The first 1.6 experimental release of Windows App SDK is now available! Releases in the experimental channel include features that are in the early stages of development. Experimental features may be removed from the next release or may never be released. See [Windows App SDK release channels](https://docs.microsoft.com/windows/apps/windows-app-sdk/release-channels) for more information about the various release channels.

This release includes bug fixes from 1.5, access to non-stable APIs, and new features including AOT support, improved TabView tab tear-out, and a new TitleBar control.

**Check out the [release notes](https://aka.ms/windowsappsdk/1.6/1.6-experimental1/release-notes) for all of the new and updated features, limitations, and known issues.**

### Updating to this version

The 1.6.240531000-experimental1 package is available at: https://www.nuget.org/packages/Microsoft.WindowsAppSDK/1.6.240531000-experimental1

Additionally, for those testing deployment of unpackaged apps, download the Windows App SDK Installers and MSIX packages [here](https://aka.ms/windowsappsdk/1.6/1.6.240531000-experimental1/Microsoft.WindowsAppRuntime.Redist.1.6.zip).

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/WindowsAppSDK/releases/tag/v1.6-exp1'>v1.6-experimental1</a>.</em>

**Comment by christosk92** [comment]
Great stuff again. I love the work the team has been putting in for the WinAppSdk. I'm sure it has a bright future ahead.

**Comment by Balkoth** [comment]
Enabling `PublishAot` like in the docs does the following to my app. How is anyone supposed to figure out what is going on here and how to fix it? Despite this being built and run on the same machine with symbols available in Visual Studio 2022 17.10.1 i can only navigate to the disassembly.

![image](https://github.com/microsoft/WindowsAppSDK/assets/80055480/8fb18245-e747-4f81-8b49-b532575ea9f3)

**Reply by Scottj1s** [reply]
@Balkoth This issue appears to be unrelated to Windows App SDK, as a module supporting System.Printing cannot be found.  [Process Monitor](https://learn.microsoft.com/en-us/sysinternals/downloads/procmon) might help track down the missing module and the locations being probed.  Otherwise, you may need to open an issue within Visual Studio.

Also Note: With PublishAot, the app is a native program.  Visual Studio F5 on a csproj with AOT is (currently) unaware of this and attempts to attach the managed debugger, which fails.  To work around this, a startup debugger hook can be added, the app launched via Ctrl+F5, and then VS attached with the native debugger.  Any attach combination involving the managed debugger will fail.  Note that C# source-level debugging works with the native debugger.  Tracking issue: [For a C# project with PublishAot=true, Visual Studio attempts to launch the managed debugger, which fails - Developer Community](https://developercommunity.visualstudio.com/t/For-a-C-project-with-PublishAottrue-V/10654472?port=1025&fsid=278aa8bf-b0aa-43f3-87bd-2acd03650b6c)

**Comment by kpavlis** [comment]
I can't change the ElementTheme of the MainWindow and other elements during Runtime (for example, when the user wants to switch to Dark Mode via a button click).

I have written this code :

 **grid_obg.RequestedTheme = ElementTheme.Dark; **

...and there is no change. 

In previous WinAppSDK versions, it was working properly changing the theme.

**Reply by kpavlis** [reply]
@Scottj1s Hi, are you aware of that bug ;

**Comment by BlameTwo** [comment]
I have a question
I performed AOT as mentioned above the document. The subsequent project was generated normally. I did not use PowerShell, but I used the side load of Visual Studio 2022. of.
![278a5cce476d9c1df35a9939d856bd9c](https://github.com/microsoft/WindowsAppSDK/assets/65759143/370a6d8f-b95b-4cfd-9e65-04f74d196146)

**Reply by Scottj1s** [reply]
Hi @BlameTwo, what's your question regarding AOT?

**Comment by BlameTwo** [comment]
This is my project file

![image](https://github.com/microsoft/WindowsAppSDK/assets/65759143/d53d13ad-bcfc-421c-8291-41b28f786dee)

**Reply by Scottj1s** [reply]
The [release notes](https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/experimental-channel#native-aot-support) are a bit confusing on this point - we'll clarify them. PublishAot must be set to true at nuget restore time, if there is any desire to use it at build/package time (e.g., in a publish profile for Release builds only).  In your case, you're only setting it at nuget restore time.  You can remove the condition, or set PublishAot elsewhere for build time.

**Reply by Poker-sang** [reply]
@Scottj1s Thanks. And I found that using the command line `dotnet publish -p:Platform=x64 -p:GenerateAppxPackageOnBuild=true` I can publish correctly AOT , but if I use the package and publish in VS2022 I can't build msix successfully, is this normal?

**Reply by Scottj1s** [reply]
> @Scottj1s Thanks. And I found that using the command line `dotnet publish -p:Platform=x64 -p:GenerateAppxPackageOnBuild=true` I can publish correctly AOT , but if I use the package and publish in VS2022 I can't build msix successfully, is this normal?

Poker-sang, can you provide any error messages when you try to package and publish in VS?  Or a minimal repro project?

**Reply by Poker-sang** [reply]
> Poker-sang, can you provide any error messages when you try to package and publish in VS? Or a minimal repro project?

Sorry, I found the problem solved and I appreciate the heads up. I realized it was probably because I needed to set pubxml to `<PublishTrimmed>True</PublishTrimmed>` to get it to work.

**Comment by ghord** [comment]
Could we get a sample C# hello world Unpackaged application that publishes as Aot from Visual Studio? I've been struggling to get anything running, my published app simply crashes at startup.

**Reply by ghost1372** [reply]
AOT is not ready yet. You can find some notes about building aot In wasdk v1.6-exp1 release note. Wait for exp2 or preview1.
